From ac5608c300e2f3c1ee39c9a9da895b43573e2fb9 Mon Sep 17 00:00:00 2001
From: Ong Boon Leong <boon.leong.ong@intel.com>
Date: Mon, 1 Jun 2020 18:05:52 +0800
Subject: [PATCH 65/77] net: stmmac: clear xdpf after TX buffer cleaned

Fix an issue where xdpf is not cleared during stmmac_clean_tx_queue.

Signed-off-by: Ong Boon Leong <boon.leong.ong@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_xsk.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_xsk.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_xsk.c
index 5303273b31bf..9e0f6a34889b 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_xsk.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_xsk.c
@@ -1062,6 +1062,7 @@ void stmmac_xsk_clean_tx_queue(struct stmmac_tx_queue *tx_q)
 		else
 			xsk_frames++;
 
+		tx_q->xdpf[ntc] =  NULL;
 		ntc = STMMAC_GET_ENTRY(ntc, priv->dma_tx_size);
 	}
 
-- 
2.17.0

